tests/structured_conf/test_structured_config.py

```diff
@@ -787,6 +787,7 @@ def test_assign_wrong_type_to_dict(self, class_type: str, value: Any) -> None:
         cfg = OmegaConf.structured(module.ConfigWithDict2)
         with pytest.raises(ValidationError):
             cfg.dict1 = value
+        cfg == OmegaConf.structured(module.ConfigWithDict2)
```

Comment:
I've just noticed that the content_backup was wrong and I didn't see this stupid line of code. I'll check it tomorrow got a test test first in the morning.

Reply:
oops.

Reply:
I fixed it, the exception is picked at `__setitem__` so it made sense to restore the backup there. I've made previous_content assigning _content to it instead of copying.

Reply:
check this